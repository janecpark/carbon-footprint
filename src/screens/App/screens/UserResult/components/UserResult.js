import React, { useContext, useEffect, useState } from 'react';
import CarbonApi from '../../../CarbonApi';
import UserContext from '../../../UserContext';
import ResultCard from './ResultCard';

/** Fetches user result from database and renders a list */

const UserResult = () => {
  const { curUser } = useContext(UserContext);
  const [userData, setUserData] = useState([]);
  useEffect(() => {
    async function userResult() {
      if (curUser) {
        const result = await CarbonApi.getResult(curUser);
        setUserData(result.data.result);
      }
    }
    userResult();
  }, [curUser]);

  async function removeResult(idx) {
    await CarbonApi.removeResultApi(idx);
    setUserData(userData.filter((r) => r.id !== idx));
  }

  const displayResult = userData.map((r) => (
    <ResultCard
      key={r.id}
      id={r.id}
      food={r.food}
      housing={r.housing}
      total={r.total}
      date={r.posting_date}
      transport={r.transport}
      removeResult={removeResult}
    />
  ));

  return (
    <div className="User-result">
      <h2 className=" User-result-text text-center container">Saved Results</h2>
      {userData.length ? 
      <div className="card-container">{displayResult}</div> : <p className='text-center'> No saved results! </p>
      }
    </div>
  );
};

export default UserResult;
