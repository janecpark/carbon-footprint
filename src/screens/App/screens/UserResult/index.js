import UserResult from './components/UserResult';

export default UserResult;