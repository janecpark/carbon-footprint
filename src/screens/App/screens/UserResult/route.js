import { routeConstants } from '../../../../shared/constants';
import UserResult from "./";

export default {
    path: routeConstants.USER_RESULT.route,
    component: UserResult
};