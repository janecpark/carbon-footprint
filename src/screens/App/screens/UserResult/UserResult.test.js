import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import UserResult from './components/UserResult';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <UserResult />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <UserResult />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

