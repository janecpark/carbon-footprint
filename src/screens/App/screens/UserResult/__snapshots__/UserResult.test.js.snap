// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="User-result"
  >
    <h2
      class=" User-result-text text-center container"
    >
      Saved Results
    </h2>
    <p
      class="text-center"
    >
       No saved results! 
    </p>
  </div>
</DocumentFragment>
`;
